server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/visitors/CloudFoundryNodeVisitor.java

```diff
@@ -0,0 +1,107 @@
+package org.opentosca.toscana.plugins.cloudfoundry.visitors;
+
+import org.opentosca.toscana.model.node.Apache;
+import org.opentosca.toscana.model.node.Compute;
+import org.opentosca.toscana.model.node.MysqlDatabase;
+import org.opentosca.toscana.model.node.MysqlDbms;
+import org.opentosca.toscana.model.node.RootNode;
+import org.opentosca.toscana.model.node.WebApplication;
+import org.opentosca.toscana.model.operation.Operation;
+import org.opentosca.toscana.model.operation.OperationVariable;
+import org.opentosca.toscana.model.visitor.StrictNodeVisitor;
+import org.opentosca.toscana.plugins.cloudfoundry.CloudFoundryApplication;
+
+import static org.opentosca.toscana.plugins.cloudfoundry.CloudFoundryManifestAttribute.DISKSIZE;
+import static org.opentosca.toscana.plugins.cloudfoundry.CloudFoundryManifestAttribute.DOMAIN;
+import static org.opentosca.toscana.plugins.cloudfoundry.CloudFoundryManifestAttribute.MEMORY;
+
+public class CloudFoundryNodeVisitor implements StrictNodeVisitor {
+
+    private CloudFoundryApplication myApp;
+
+    public CloudFoundryNodeVisitor(CloudFoundryApplication myApp) {
+        this.myApp = myApp;
+    }
+
+    public CloudFoundryApplication getFilledApp() {
+        return myApp;
+    }
+
+    @Override
+    public void visit(Compute node) {
+        /*
+        os: do nothing -> CF uses linux based containers
+        private_address: do nothing -> automatically
+        public_address: add value to manifest
+        networks: do nothing -> automatically
+        ports: do nothing -> automatically
+        */
+
+        if (node.getPublicAddress().isPresent()) {
+            myApp.addAttribute(DOMAIN.getName(), node.getPublicAddress().get());
+        }
+
+        if (node.getHost().getDiskSizeInMB().isPresent()) {
+            myApp.addAttribute(DISKSIZE.getName(), node.getHost().getDiskSizeInMB().get() + " MB");
+        }
+
+        if (node.getHost().getMemSizeInMB().isPresent()) {
+            myApp.addAttribute(MEMORY.getName(), node.getHost().getMemSizeInMB().get() + " MB");
+        }
+    }
+
+    @Override
+    public void visit(MysqlDatabase node) {
+        /*
+        create service
+        ignore password and port
+         */
+        myApp.addService(node.getNodeName());
+    }
+
+    @Override
+    public void visit(MysqlDbms node) {
+        // TODO: check how to configure database
+        handleStandardLifecycle(node);
+    }
+
+    @Override
+    public void visit(Apache node) {
+        // do nothing yet but override to prevent a UnsupportedTypeException
+    }
+
+    @Override
+    public void visit(WebApplication node) {
+        myApp.setName(node.getNodeName());
+        handleStandardLifecycle(node);
+    }
+
+    public void handleStandardLifecycle(RootNode node) {
+        for (Operation operation : node.getStandardLifecycle().getOperations()
+            ) {
+            if (operation != null) {
```

Comment:
I've fixed the bug [leading to null elements in the operation set](in 11020c28a0b643688f6726121ed1789c84c18d7d).
I suppose you dismiss the null-check.